diff --git a/src/qt_gui/main.cpp b/src/qt_gui/main.cpp
index 68a674f5..a6331fc8 100644
--- a/src/qt_gui/main.cpp
+++ b/src/qt_gui/main.cpp
@@ -31,8 +31,10 @@ int main(int argc, char* argv[]) {
     qInstallMessageHandler(customMessageHandler);
 
     // Initialize the main window
-    MainWindow* m_main_window = new MainWindow(nullptr);
-    m_main_window->Init();
+    if (!has_command_line_argument) {
+        MainWindow* m_main_window = new MainWindow(nullptr);
+        m_main_window->Init();
+    }
 
     // Check for command line arguments
     if (has_command_line_argument) {
